RocketChat/Rocket.Chat.ReactNative
# Chore: Hooks App/share

## Proposed changes

Migrate the app/share.tsx from Class to Hooks. And cleaned some functions that aren't in use.

## Issue(s)	


## How to test or reproduce


- Open the gallery or folder from your smartphone 
- Share something by Rocket.Chat
- Should open the Share Extension
- Try logged in some server and without logged

## Screenshots

## Types of changes




## Checklist



## Further comments
